stac_fastapi/eodag/core.py

```diff
@@ -279,31 +283,32 @@ async def all_collections(
                 guessed_collections = request.app.state.dag.guess_collection(
                     free_text=free_text, start_date=start, end_date=end
                 )
+                guessed_collections_ids = [coll.id for coll in guessed_collections]
             except EodagNoMatchingCollection:
-                collections = []
+                collections = CollectionsList([])
             else:
-                collections = [pt for pt in all_pt if pt["ID"] in guessed_collections]
+                collections = CollectionsList([pt for pt in all_pt if pt.id in guessed_collections_ids])
```

Comment:
```suggestion
                collections = CollectionsList([c for c in all_colls if c.id in guessed_collections_ids])
```
also rename `all_pt` to `all_colls` just above